scripts/ci/remote-build-artifacts.sh

```diff
@@ -0,0 +1,184 @@
+    # Other actors (repositories) can easily see the flags.
+    # When `mapbox-vision-ios` repo wants to pull the new verison of framework, it does the following:
+    # 1. raise the flag (creates a pullMarkerFlag file)
+    # 2. when there's no syncMarkerFlag, we start pulling process
+    # 3. when the pulling is completed, we lower the flag (remove pullMarkerFlag file)
+    #
+    # What we do in case when there's a syncMarkerFlag (step 2 above)?
+    # - we lower our flag (removes pullMarkerFlag file)
+    # - we wait until sync flag is lowered (syncMarkerFlag is removed)
+    # - then we raise our flag (create pullMarkerFlag file)
+    #
+    # The same behaviour is applicable for other actors (including mapbox-vision repo).
+
+    echo
+    echo "Checking if build artifacts are currently being updated"
+    echo
+    syncMarkerFilePath="${s3BuildDir}/${frameworkName}/${syncMarkerFileName}"
+    pullMarkerFilePath="${s3BuildDir}/${frameworkName}/${pullMarkerFileName}"
+
+    while : ; do
+        echo "Putting a pull marker to indicate we want to pull latest build artifacts"
+        createEmptyFileOnS3 "${pullMarkerFilePath}"
+
+        if isFileExistsOnS3 "${syncMarkerFilePath}"; then
+            echo "Sync marker was found. That means uploading of build artifacts is in progress"
+            echo "Removing the pull marker"
+            aws s3 rm "${pullMarkerFilePath}" --quiet
+        else
+            echo "Sync marker was not found. That means we can start pulling process"
+            break
+        fi
+
+        echo "Retrying in several seconds"
+        echo
+        sleep $[ ( $RANDOM % 20 )  + 1 ]s # 1-20 seconds random sleep, time interval to check flag next time
+    done
+
+    echo
+    echo "Start syncing"
+    echo
+
+    aws s3 sync "${s3BuildDir}/${frameworkName}/${framework}" "${localFrameworkPath}"
+
+    echo
+    echo "Removing the pull marker to allow external repos to upload updated build artifacts"
+    echo
+    aws s3 rm "${pullMarkerFilePath}" --quiet
+
+    performCleanup
+    echo "${framework} was sucсessfully pulled"
+}
+
+function pullNativeBuildProducts() {
+    if [[ $# -ne 0 ]]; then
+        echoerr "Illegal arguments passed to ${FUNCNAME[0]}"
+        exit "$ERROR_ILLEGAL_NUMBER_OF_ARGS"
+    fi
+
+    local -r buildArtifactsToPull=("MapboxVisionNative" "MapboxVisionARNative" "MapboxVisionSafetyNative")
+
+    for buildArtifact in "${buildArtifactsToPull[@]}"; do
+        pullFramework "${buildArtifact}"
+
+        # copy pulled frameworks into Carthage/Build/iOS dir to allow building
+        echo
+        echo "Copying ${buildArtifact} into Carthage/Build/iOS dir"
+        cp -a "Carthage/Build/iOS/Release-${IOS_PLATFORM_TYPE}/${buildArtifact}.framework" "Carthage/Build/iOS"
+    done
+    
+    echo
+    echo "Build dependecies are up-to-date now."
+    echo
+}
+
+###################
+#   Main script   #
+###################
+
+if [[ $# -gt 0 ]]; then
+    while [[ $# -gt 0 ]]; do
```

Comment:
nit:
```suggestion
    for argument in "$@" do
```
and without `shift` command below